src/core/render/compiler/blockquote.js

```diff
@@ -0,0 +1,36 @@
+export const blockquoteCompiler = ({ renderer }) =>
+  (renderer.blockquote = function ({ tokens }) {
+    const calloutData =
+      tokens[0].type === 'paragraph' &&
+      // 0: Match "[!TIP] My Title"
+      // 1: Mark "[!TIP]"
+      // 2: Type "TIP"
+      tokens[0].raw.match(/^(\[!(\w+)\])/);
```

Comment:
```suggestion
      tokens[0].raw.match(/^(\[!(CAUTION|IMPORTANT|NOTE|TIP|WARNING)\])/);
```

I think we need strict match the callout tags in case of users use something like `![DUCK]` in some reason.

Reply:
This was intentional. :)

Callouts (or "admonitions" or "alerts") are found in many other apps and the list of callouts supported is often much larger than what GitHub supports. By capturing the string contained with the brackets (`[!STRING]`) and using it as a callout class (`<div class="callout string">`) it easy for third parties to expand the number of callouts supported. 

For example:

```markdown
> [!BUG]
> This is a bug callout
```

```css
.callout.bug {
  --callout-bg: orange;
}
```

Reply:
I see. so it means the `![words]` is a `callout` syntax hook in docsify and users could do the extension.

```
> [!question] Should we mentione it in `UI-KIT`?
> > [!advantage] users can understand there is a way to do extension for callout block.
> > > [!example]  or we could also add a example.
```

Reply:
Correct.

The "callout" class handles all of the base styling. The additional class on the callout element ("caution", "important", "note", etc.) is used to adjust styles like colors and icons. To extend the system to support the example you provided above ("question", "advantage", "example") they would simply need to create CSS declarations for each:

```css
callout.question {
  --callout-bg: red;
  --callout-color: white;
}
```

I can add a note about custom classes to the docs along with a note explaining that in most cases custom classes will not work outside of Docsify.

Reply:
great!

Reply:
> I can add a note about custom classes to the docs along with a note explaining that in most cases custom classes will not work outside of Docsify.

Hi, @jhildenbiddle  Do you still need to modify the document? If not, then you can merge this PR.

Reply:
@sy-records Yep. Haven't had a chance to complete the updated docs, but I'll have that in the next or two.

Reply:
@Koooooo-7 @sy-records --

One other enhancements I'd like to make is the ability to define localized labels. I make a conscious decision to not implement this initially for a few reasons, but after thinking it over for a bit I've decided it makes sense to offer this feature.

The implementation will follow the same pattern we've used elsewhere in our configuration. The default behavior will be to not render text labels (as shown in screenshots above):

```js
window.$docsify = {
  // Default: do not render text labels
  callouts: {},
};
```

If users want to render text labels automatically as is done in markdown environments like GitHub and Obsidian, they can define them as part of their configuration:

```js
// Default values
window.$docsify = {
  // Render same label for all route paths
  callouts: {
    caution: 'Caution',
    important: 'Important',
    note: 'Note',
    tip: 'Tip',
    warning: 'Warning',
  },
};
```

```js
window.$docsify = {
  // Render localized labels based on route paths
  callouts: {
    caution: {
      '/es/': 'Precaución',
      '/de-de/': 'Vorsicht',
      '/ru-ru/': 'Осторожность',
      '/zh-cn/': '警告',
      '/': 'Caution',
    },
    // ...
  },
};
```

Reply:
----- @jhildenbiddle , Currently, we simply add a icon per callout config. That looks good to me.
![TIP](https://github.com/user-attachments/assets/22cf63f3-e21b-4885-a097-e7b4f099b4a9)


I don't propose that we need introduce a `label` into docsify:
- We need extract the `content` from the `[!HERE]` and match it to the `$.callouts` config by path and by content.
   i.e. if user has the `![example]`, they absolutely want to config it like this:
```js
window.$docsify = {
  // Render same label for all route paths
  callouts: {
    example: {
      '/zh-cn/': '例子',
      '/': 'Example',
    },
  },
```

It makes the config more complex and messy,  they need config options size is:  `i18n-paths * callouts`.


IMO, the `icon` already has the `info`, we don't need mention it specially with a `label`.
In a common sense, a light lamp is `tips`, a `i` is info/note, a red thing is forbidden/attention, a yellow triangle is warning.


The only different thing is we introduce some `icon` more docsify style (such as `important` is a cute star.). thats fine since we don't fu*k any things up. 

If user do wanna enrich the `label` in their multi lang sites, they can simply add something like this: 
- `> [!IMPORTANT] **Important** in EN site.`
- `> [!IMPORTANT] **重要** in ZH site.`